src/main/scala/org/broadinstitute/dsde/agora/server/dataaccess/AgoraDao.scala

```diff
@@ -4,16 +4,33 @@ package org.broadinstitute.dsde.agora.server.dataaccess
 import com.mongodb.casbah.MongoCollection
 import org.broadinstitute.dsde.agora.server.dataaccess.mongo.AgoraMongoClient._
 import org.broadinstitute.dsde.agora.server.dataaccess.mongo.AgoraMongoDao
-import org.broadinstitute.dsde.agora.server.model.{AgoraEntity, AgoraEntityProjection}
+import org.broadinstitute.dsde.agora.server.model.{AgoraEntityType, AgoraEntity, AgoraEntityProjection}
 
 object AgoraDao {
   //factory method (currently only supports mongo)
-  def createAgoraDao: AgoraDao = {
-    new AgoraMongoDao(getMethodsCollection(getMongoClient))
+  def createAgoraDao(entityType: Option[AgoraEntityType.EntityType]): AgoraDao = {
+    new AgoraMongoDao(getCollectionsByEntityType(entityType))
   }
 
   def createAgoraDao(collection: MongoCollection): AgoraDao = {
-    new AgoraMongoDao(collection)
+    new AgoraMongoDao(Seq(collection))
+  }
+
+  def createAgoraDao(entityTypes: Seq[AgoraEntityType.EntityType]): AgoraDao ={
+    new AgoraMongoDao((for(entityType <- entityTypes) yield getCollectionsByEntityType(Option(entityType))).flatten)
+  }
+
+  def getCollectionsByEntityType(entityType: Option[AgoraEntityType.EntityType]) : Seq[MongoCollection] ={
```

Comment:
Does this belong here, or in the client with the other MongoCollection returning functions?